packages/react-workers/src/workerPool.js

```diff
@@ -0,0 +1,62 @@
+import viewportFeaturesWorker from './workers/viewportFeatures.worker';
+
+const pool = {};
+
+export function executeTask(source, method, params) {
+  return new Promise((resolve, reject) => {
+    const worker = getWorker(source);
+    worker.tasks.push({
+      method,
+      params,
+      resolve,
+      reject
+    });
+    if (worker.tasks.length === 1) {
+      resolveWorkerTasks(worker);
+    }
+  });
+}
+
+export function removeWorker(source) {
+  if (pool[source]) {
+    pool[source].tasks.forEach((t) => t.reject());
+    pool[source].worker.terminate();
+    delete pool[source];
+  }
+}
+
+function getWorker(source) {
+  if (!pool[source]) {
+    pool[source] = {
+      worker: new viewportFeaturesWorker(),
+      tasks: []
+    };
+    onmessage(pool[source]);
```

Comment:
Maybe rename to addOnMessageHandler / addOnErrorHandler ?

Also I would use const worker = pool[source] as the first line, and then substitute other calls, if possible, with just 'worker'

Reply:
In this case, I prefer the native names.

`Also I would use const worker = pool[source] as the first line, and then substitute other calls` I think you're losing your mind with the Clean Code course XD. Seriously, that change doesn't add any clarity to the code.